### PR TITLE
Terraform dashboard

### LINKED ISSUE #39: Terraform Script for Dashboard
### Steps

- [x] Write Terraform scripts to provision necessary AWS resources (ECS).
- [x] Test Terraform scripts in a staging environment.

### User Story

As a DevOps engineer, I want to use Terraform to automate the infrastructure setup so that the deployment process is reproducible and scalable.

### PR DESCRIPTION
**Pull request by BerkayDur**
Terraform the dashboard
closes #39

### CODE REVIEW

File: Dashboard/Terraform/main.tf [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+        {
+            cidr_blocks      = ["0.0.0.0/0"]
+            description      = ""
+            from_port        = var.PORT_TO_EXPOSE
+            ipv6_cidr_blocks = []
+            prefix_list_ids  = []
+            protocol         = "tcp"
+            security_groups  = []
+            self             = false
+            to_port          = var.PORT_TO_EXPOSE
```
Comment (BerkayDur) [inline Dashboard/Terraform/main.tf:28]: Nice use of variables.

File: Dashboard/Terraform/main.tf [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+            memory = 3072
+            essential = true
+            portMappings = [
+                {
+                    hostPort = 80
+                    containerPort = 80
+                },
+                {
+                    hostPort = 8501
+                    containerPort = 8501
```
Comment (BerkayDur) [inline Dashboard/Terraform/main.tf]: Could use variables here but this is optional as users will interact with this port

**Review by BerkayDur — COMMENTED** [review]
Looks good, just a few suggestions!

File: Dashboard/Terraform/main.tf (hunk 2 above)
Comment (danfh00) [reply Dashboard/Terraform/main.tf]: done

File: Dashboard/utilities/extract_combined.py
Comment (A-Faris) [inline Dashboard/utilities/extract_combined.py:1]: Delete this if it is not needed

File: Dashboard/Terraform/variables.tf [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+}
+
+variable "AWS_REGION" {
+    type = string
+    default = "eu-west-2"
+}
+
+variable "dashboard_name" {
+  default = "c11-hermes-dashboard-tf-1"
+}
```
Comment (A-Faris) [inline Dashboard/Terraform/variables.tf]:
```suggestion
variable "dashboard_name" {
    type = string
    default = "c11-hermes-dashboard-tf-1"
}
```

File: Dashboard/Terraform/variables.tf [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+variable "DASHBOARD_NAME" {
+    type = string
+}
+
+variable "CLUSTER_ARN" {
+    type = string
+}
+
+variable "SUBNETS" {
+    type=list(string)
```
Comment (A-Faris) [inline Dashboard/Terraform/variables.tf]:
```suggestion
    type = list(string)
```

File: Dashboard/Terraform/variables.tf (hunk 3 above)
Comment (danfh00) [reply Dashboard/Terraform/variables.tf]: done

File: Dashboard/Terraform/variables.tf (hunk 4 above)
Comment (danfh00) [reply Dashboard/Terraform/variables.tf]: done

File: Dashboard/Terraform/variables.tf [hunk 5]
```diff
@@ -0,0 +1,57 @@
+
+
```
Comment (A-Faris) [inline Dashboard/Terraform/variables.tf:2]: remove these 2 lines

**Review by A-Faris — APPROVED** [review]
All great thanks

**Pull request merged**